Elixir examples based on @lexmag Elixir Style Guide

Eiji7 opened the pull request:
### What does this implement/fix? Explain your changes.
Examples for [elixir](https://github.com/elixir-lang/elixir) (Elixir language).
Elixir Original <=> Expected examples based on @lexmag https://github.com/lexmag/elixir-style-guide
For more information see related issue: #545.

### Checklist

Check all those that are applicable and complete.

- [x] Merged with latest `master` branch
- [x] Added examples for testing to [examples/ directory](examples/)
- [ ] Travis CI passes (Mac support)
- [ ] AppVeyor passes (Windows support)

lexmag commented on examples/nested-jsbeautifyrc/elixir/expected/lexmag_2_syntax_09_true_in_cond.ex:
```diff
@@ -0,0 +1,37 @@
+      char in ?A..?Z ->
+        char - ?A + 10
+      true ->
+        nil # NOTE: beautifier info to fill: "`true` as the always-match condition"
```
I'm not sure why we can use `char - ?a + 10` here instead of `nil`?

lexmag replied:
Now I see, the rule wasn't clear, it meant the we should always use `true` and not any other atom since `cond` allows any:
```elixir
cond do
  :any -> :ok
end
```

Eiji7 replied:
@lexmag:
I used nil before comment to not display warning on compilation.
It's example value that developer should replace with comment.
Should I change `true` to `:any`?

lexmag commented on examples/nested-jsbeautifyrc/elixir/expected/lexmag_2_syntax_10_boolean_operators.ex:
```diff
@@ -0,0 +1,33 @@
+  @spec test(name :: atom) :: any
+  def test(name) do
+    task = :spying
+    is_atom(name) and !is_nil(name) and !is_boolean(name)
```
We should not use `!` with strictly boolean values.

Eiji7 replied:
right, my bad

lexmag commented on examples/nested-jsbeautifyrc/elixir/expected/lexmag_3_naming_3_predicate_funs_name.ex:
```diff
@@ -0,0 +1,36 @@
+  def ok?(_), do: false
+
+  @spec is_ok?(sample :: any) :: boolean
+  defmacro is_ok?(:ok), do: true
```
We should not use `?` if we have `is_` prefix.

Eiji7 replied:
right, my bad

lexmag commented on examples/nested-jsbeautifyrc/elixir/expected/lexmag_4_comments_3_no_superfluous_comments.ex:
```diff
@@ -0,0 +1,25 @@
+  """
+  @spec test() :: any
+  def test() do
+    # capitalize sentence
```
I assume we can't validate this rule, right?

Eiji7 replied:
hmm, how about to add:
```elixir
# beautifier NOTE: is this comment: "comment content", really needed?
```

lexmag commented:
I think we can replace Markdown fences ("\```") with 4 spaces indentation, it is more common way to format code examples inside `@doc """` content.

Eiji7 commented on examples/nested-jsbeautifyrc/elixir/expected/lexmag_2_syntax_03_pipeline_operator.ex:
```diff
@@ -0,0 +1,50 @@
+    " EXAMPLE "
+    |> String.strip
+    |> String.downcase
+    String.upcase("example")
```
ok

Eiji7 commented on examples/nested-jsbeautifyrc/elixir/expected/lexmag_2_syntax_06_for_indentation.ex:
```diff
@@ -0,0 +1,53 @@
+    end
+
+    indexes = 0..999
+    max_count =
```
ok

Eiji7 commented:
@lexmag: Thanks for you review! I corrected all places you selected and changed Markdown fences to 4 spaces. We only need to think about new issue I referenced.

Glavin001 commented:
This looks great to me! Great work! I can see the tests are all failing for Elixir, as expected. However, now we need to add an Elixir beautifier such that they pass. Did someone from https://github.com/Glavin001/atom-beautify/issues/545 find a beautifier or is going to implement one? /cc @prettydiff

Eiji7 commented:
@Glavin001: I don't know if there is any beautifier that uses rules from this style guide, but if you want a example code to analysis Elixir projects then [credo](http://credo-ci.org/) might help you. 
**Note**: Credo uses its own [style guide](https://github.com/rrrene/elixir-style-guide).

Eiji7 commented:
@Glavin001: I found [linter-elixir-credo](https://atom.io/packages/linter-elixir-credo) Atom package that could help you with implementing beautifier.

Glavin001 commented:
Just a quick note: since these tests currently fail the build -- as expected -- we are unable to merge until a suitable Elixir beautifier is found and integrated into Atom-Beautify. Until then, I will keep this Pull Request open as it is great work and will help us thoroughly test the Elixir beautifier when it is found. Thank you.

Glavin001 commented:
Cleaning up PRs. Can be reopened later once a Elixir beautifier is found and these can be used for testing 😃 .
